website/mails/mails.py

```diff
@@ -232,3 +312,35 @@ def send_mail(to_addr, mail, mimetype='plain', from_addr=None, mailer=None,
     'welcome',
     subject='Welcome to the Open Science Framework'
 )
+
+# Predefined emails with callbacks
+NO_ADDON = {
+    'template': 'no_addon',
+    'subject': 'Link an add-on to your OSF project',
+    'callback': mail_callbacks.no_addon
```

Comment:
Discussed in person: `presend` would be a better name for these. Also, document (perhaps ~L316, above all these constants) the semantics of these dictionaries. For example, you should document that `presend` returns a boolean (whether the email should be sent) and also may have side effects (as is the case with `welcome_osf4m`.